### PR TITLE
[codex] Fix scrollable reader tables

### LINKED ISSUE #72: tables are malformed and not horizontally scrollable
currently the table overflow and make the whole webview wider than the screen, which makes the webview horizontally scrollable and is a bad ux. What i want is to make the table itself horizontally scrollable when it's wider than the screen.

This example is from windscribe RSS feed (https://windscribe.com/blog/rss/).

### PR DESCRIPTION
**Pull request by johan456789**
Closes #72

## Summary
- wrap article tables in a WebView scroll container so wide tables scroll horizontally on their own
- preserve normal table layout instead of forcing rows into block layout
- add an emulator-backed WebView regression test for wide table overflow

## video

https://github.com/user-attachments/assets/f36b5178-3d4c-40a5-95ee-8ec5c1b56558

## Validation
- ./gradlew :app:compileGithubDebugKotlin :app:compileGithubDebugAndroidTestKotlin
- ./gradlew :app:connectedGithubDebugAndroidTest -Pandroid.testInstrumentationRunnerArguments.class=me.ash.reader.ui.component.webview.RYWebViewTableTest
- Manual emulator smoke test: installed and launched githubDebug for visual testing

### CODE REVIEW

**Comment by johan456789** [comment]
@codex review

File: app/src/main/java/me/ash/reader/ui/component/webview/WebViewScript.kt [hunk 1]
```diff
@@ -77,6 +77,17 @@ function setBold(enabled) {
 
 ${if (boldCharacters) "setBold(true);" else ""}
 
+document.querySelectorAll("table").forEach(function(table) {
+    if (table.closest(".table-scroll") || table.parentElement.closest("table")) {
```
[the remark this answers is not in the document]
Comment (johan456789) [reply app/src/main/java/me/ash/reader/ui/component/webview/WebViewScript.kt]: cdc3d01

File: app/src/androidTest/java/me/ash/reader/ui/component/webview/RYWebViewTableTest.kt [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        var encodedResult: String? = null
+        Handler(Looper.getMainLooper()).post {
+            webView!!.evaluateJavascript(script) {
+                encodedResult = it
+                latch.countDown()
+            }
+        }
+        assertTrue(latch.await(5, TimeUnit.SECONDS))
+        val json = JSONTokener(encodedResult).nextValue() as String
+        return org.json.JSONObject(json)
```
[the remark this answers is not in the document]
Comment (johan456789) [reply app/src/androidTest/java/me/ash/reader/ui/component/webview/RYWebViewTableTest.kt]: [d70a1e7](https://github.com/johan456789/ReadYou/pull/73/commits/d70a1e73cce7d4f5d6e1f47db05e9082b4a6ba9e)

File: app/src/main/java/me/ash/reader/ui/component/webview/WebViewStyle.kt [hunk 3]
```diff
@@ -248,13 +248,22 @@ blockquote img {
...
+    display: block !important;
+    max-width: 100% !important;
+    overflow-x: auto !important;
+    overflow-y: hidden !important;
+    margin-top: 1em !important;
+    margin-bottom: 1em !important;
+    -webkit-overflow-scrolling: touch;
+}
+
+.table-scroll table {
```
[the remark this answers is not in the document]
Comment (johan456789) [reply app/src/main/java/me/ash/reader/ui/component/webview/WebViewStyle.kt]: [e6a6602](https://github.com/johan456789/ReadYou/pull/73/commits/e6a6602132f9aa7f619ee7ad950575c7c56e92e0)
